<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">

    

    <title>
      融合基因 | Sunshine&#39;s Life 
    </title>

    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
    
      <meta name="author" content="杨光">
    
    

    <meta name="description" content="添加融合基因分析内容tophat-fusion软件分析说明TopHat-Fusion可以从单端或双端的RNA-seq数据中识别融合基因。它将首次未比对到基因组上的reads进行打断，通过重新比对寻找候选的融合基因，后续采用多重规则进行过滤和筛选。该算法不依赖于已知的基因注释信息，这使其具有更高的敏感性，除了鉴定已知基因的融合，它还可以发现来自新基因或已知基因新剪切体的融合产物">
<meta property="og:type" content="article">
<meta property="og:title" content="融合基因 | Sunshine's Life">
<meta property="og:url" content="http://yoursite.com/2017/02/28/融合基因/index.html">
<meta property="og:site_name" content="Sunshine's Life">
<meta property="og:description" content="添加融合基因分析内容tophat-fusion软件分析说明TopHat-Fusion可以从单端或双端的RNA-seq数据中识别融合基因。它将首次未比对到基因组上的reads进行打断，通过重新比对寻找候选的融合基因，后续采用多重规则进行过滤和筛选。该算法不依赖于已知的基因注释信息，这使其具有更高的敏感性，除了鉴定已知基因的融合，它还可以发现来自新基因或已知基因新剪切体的融合产物">
<meta property="og:updated_time" content="2017-07-28T05:34:55.000Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="融合基因 | Sunshine's Life">
<meta name="twitter:description" content="添加融合基因分析内容tophat-fusion软件分析说明TopHat-Fusion可以从单端或双端的RNA-seq数据中识别融合基因。它将首次未比对到基因组上的reads进行打断，通过重新比对寻找候选的融合基因，后续采用多重规则进行过滤和筛选。该算法不依赖于已知的基因注释信息，这使其具有更高的敏感性，除了鉴定已知基因的融合，它还可以发现来自新基因或已知基因新剪切体的融合产物">
    
    
    
      <link rel="icon" type="image/x-icon" href="/favicon.png">
    
    <link rel="stylesheet" href="/css/uno.css">
    <link rel="stylesheet" href="/css/highlight.css">
    <link rel="stylesheet" href="/css/archive.css">
    <link rel="stylesheet" href="/css/china-social-icon.css">

</head>
<body>

    <span class="mobile btn-mobile-menu">
        <i class="icon icon-list btn-mobile-menu__icon"></i>
        <i class="icon icon-x-circle btn-mobile-close__icon hidden"></i>
    </span>

    

<header class="panel-cover panel-cover--collapsed">


  <div class="panel-main">

  
    <div class="panel-main__inner panel-inverted">
    <div class="panel-main__content">

        

        <h1 class="panel-cover__title panel-title"><a href="/" title="link to homepage">Sunshine&#39;s Life</a></h1>
        <hr class="panel-cover__divider" />

        

        <div class="navigation-wrapper">

          <nav class="cover-navigation cover-navigation--primary">
            <ul class="navigation">

              
                
                <li class="navigation__item"><a href="/#blog" title="" class="blog-button">首页</a></li>
              
                
                <li class="navigation__item"><a href="/about" title="" class="">关于</a></li>
              
                
                <li class="navigation__item"><a href="/archive" title="" class="">归档</a></li>
              

            </ul>
          </nav>

          <!-- ----------------------------
To add a new social icon simply duplicate one of the list items from below
and change the class in the <i> tag to match the desired social network
and then add your link to the <a>. Here is a full list of social network
classes that you can use:

    icon-social-500px
    icon-social-behance
    icon-social-delicious
    icon-social-designer-news
    icon-social-deviant-art
    icon-social-digg
    icon-social-dribbble
    icon-social-facebook
    icon-social-flickr
    icon-social-forrst
    icon-social-foursquare
    icon-social-github
    icon-social-google-plus
    icon-social-hi5
    icon-social-instagram
    icon-social-lastfm
    icon-social-linkedin
    icon-social-medium
    icon-social-myspace
    icon-social-path
    icon-social-pinterest
    icon-social-rdio
    icon-social-reddit
    icon-social-skype
    icon-social-spotify
    icon-social-stack-overflow
    icon-social-steam
    icon-social-stumbleupon
    icon-social-treehouse
    icon-social-tumblr
    icon-social-twitter
    icon-social-vimeo
    icon-social-xbox
    icon-social-yelp
    icon-social-youtube
    icon-social-zerply
    icon-mail

-------------------------------->

<!-- add social info here -->



<nav class="cover-navigation navigation--social">
  <ul class="navigation">

  
      <!-- Github -->
      <li class="navigation__item">
        <a href="https://github.com/yangguang8112" title="Handsome boy on GitHub">
          <i class='icon icon-social-github'></i>
          <span class="label">GitHub</span>
        </a>
      </li>


    <!-- China social icon -->
    
    <!--
      <li class="navigation__item">
        <a href="" title="">
          <i class='icon cs-icon-douban'></i>
          <span class="label">Douban</span>
        </a>
      </li>
    -->
      <li class="navigation__item">
        <a href="http://weibo.com/1994YG?refer_flag=1001030101_&is_hot=1" title="一起下厕所">
          <i class='icon cs-icon-weibo'></i>
          <span class="label">Weibo</span>
        </a>
      </li>

  



  </ul>
</nav>

        </div>

      </div>

    </div>

    <div class="panel-cover--overlay"></div>
  </div>
</header>

    <div class="content-wrapper">
        <div class="content-wrapper__inner entry">
            

<article class="post-container post-container--single">

  <header class="post-header">
    
    <h1 class="post-title">融合基因</h1>

    

    <div class="post-meta">
      <time datetime="2017-02-28" class="post-meta__date date">2017-02-28</time> 

      <span class="post-meta__tags tags">

          

          
             &#8226; 标签:
            <font class="tags">
              <a class="tags-link" href="/tags/生物信息学/">生物信息学</a>
            </font>
          

      </span>
    </div>
    
    

  </header>

  <section id="post-content" class="article-content post">
    <h3 id="添加融合基因分析内容"><a href="#添加融合基因分析内容" class="headerlink" title="添加融合基因分析内容"></a>添加融合基因分析内容</h3><h4 id="tophat-fusion软件分析说明"><a href="#tophat-fusion软件分析说明" class="headerlink" title="tophat-fusion软件分析说明"></a>tophat-fusion软件分析说明</h4><p>TopHat-Fusion可以从单端或双端的RNA-seq数据中识别融合基因。它将首次未比对到基因组上的reads进行打断，通过重新比对寻找候选的融合基因，后续采用多重规则进行过滤和筛选。该算法不依赖于已知的基因注释信息，这使其具有更高的敏感性，除了鉴定已知基因的融合，它还可以发现来自新基因或已知基因新剪切体的融合产物<br><a id="more"></a></p>
<h5 id="TopHat-Fusion下载"><a href="#TopHat-Fusion下载" class="headerlink" title="TopHat-Fusion下载"></a>TopHat-Fusion下载</h5><p>下载 Bowtie indexes (Bowtie1或者 Bowtie2), refGene.txt, and ensGene.txt . </p>
<p>从 blast database中下载 human_genomic<em>, other_genomic</em>, and nt* .</p>
<p>下载数据库（blastdb,ftp://ftp.ncbi.nlm.nih.gov/blast/db/)</p>
<h5 id="使用TopHat-Fusion"><a href="#使用TopHat-Fusion" class="headerlink" title="使用TopHat-Fusion"></a>使用TopHat-Fusion</h5><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#####一个例子</span></span><br><span class="line">tophat -o tophat_MCF7 -p 8 --fusion-search --keep-fasta-order --bowtie1 --no-coverage-search -r 0 \</span><br><span class="line">--mate-std-dev 80 --max-intron-length 100000 --fusion-min-dist 100000 --fusion-anchor-length 13 \</span><br><span class="line">--fusion-ignore-chromosomes chrM /path/to/h_sapiens/bowtie_index SRR064286_1.fastq SRR064286_2.fastq </span><br><span class="line"></span><br><span class="line"><span class="comment">#####参数说明</span></span><br><span class="line"></span><br><span class="line">-o：用tophat_(sample_name)作为输出目录</span><br><span class="line">-p：线程数</span><br><span class="line">--fusion-search 打开融合基因检测命令</span><br><span class="line">--bowtie1 调用bowtie1</span><br><span class="line">--coverage-search   关闭--coverage-search因为它会占用大量内存且令程序运行缓慢</span><br><span class="line">-r       inner mate distance</span><br><span class="line">--mate-std-dev  inner mate distance 的标准偏差，TopHat-Fusion利用这部分区域检测融合基因</span><br><span class="line">--fusion-min-dist 由于染色体重排产生的染色体内部融合最小距离</span><br><span class="line">-fusion-anchor-length reads比对锚定长度</span><br><span class="line"><span class="comment">#####在tophat运行之后, 运行 tophat-fusion-post 来筛选出融合候选基因</span></span><br><span class="line"></span><br><span class="line">tophat-fusion-post -p 8 --num-fusion-reads 1 --num-fusion-pairs 2 --num-fusion-both 5 /path/to/h_sapiens/bowtie_index </span><br><span class="line"></span><br><span class="line"><span class="comment">#####tophat-fusion-post可选参数</span></span><br><span class="line"></span><br><span class="line">-v/--version  打印出帮助信息和退出.  </span><br><span class="line">-p/--num-threads  线程数，默认值为1.  </span><br><span class="line">--num-fusion-reads  支持融合基因事件最少的reads数，默认值为3.</span><br><span class="line">--num-fusion-pairs  支持融合基因事件最少的双融合数，默认值为2. </span><br><span class="line">--num-fusion-both  支持融合基因事件最少的双融合数和reads数的总数，默认值为0.  </span><br><span class="line">--max-num-fusions  最大的融合基因数目，默认值为500.  </span><br><span class="line">--fusion-read-mismatches  mismatch值，默认值为2.  </span><br><span class="line">--fusion-multireads  忽略的具有多比对位点的Reads 默认值为2.  </span><br><span class="line">--non-human  注释的为非人的项目使用此参数.  </span><br><span class="line">--fusion-pair-dist  特定的融合距离，默认值为250.</span><br></pre></td></tr></table></figure>
<h4 id="SOAPfuse分析软件说明"><a href="#SOAPfuse分析软件说明" class="headerlink" title="SOAPfuse分析软件说明"></a>SOAPfuse分析软件说明</h4><p>SOAPfuse算法首先通过比对到基因组和转录本中双末端（pair end）关系的序列寻找候选的基因融合，然后采用局部穷举算法和一系列精细的过滤策略，在尽量保留真实融合的情况下过滤掉其中假阳性的基因融合。同时，该算法还具有融合断点预测和可视化功能，这对临床分子分型和肿瘤新药的开发具有重要意义。</p>
<h5 id="软件安装"><a href="#软件安装" class="headerlink" title="软件安装"></a>软件安装</h5><p>1.下载软件包<a href="http://soap.genomics.org.cn/soapfuse.html" target="_blank" rel="external">http://soap.genomics.org.cn/soapfuse.html</a><br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ tar -xzf /PATH_WHERE_YOU_PUT_THE_TARBALL/SOAPfuse-vX.X.tar.gz</span><br><span class="line">$ <span class="built_in">cd</span> SOAPfuse-vX.X/</span><br></pre></td></tr></table></figure></p>
<h5 id="Prepare"><a href="#Prepare" class="headerlink" title="Prepare"></a>Prepare</h5><p>1.准备样品列表</p>
<p>基于下面的格式列表文件准备样品(四列)。</p>
<p>|1|2|3|4|<br>|[sample_ID]|[sequence_library_ID]|[run_ID]|[read_length]|</p>
<p>2.准备RNA-Seq数据</p>
<p>根据样品列表存储fastq/fasta格式文件</p>
<p>构建目录存储RNA-Seq数据的五个要求:</p>
<p>（1）在主目录下存储所有RNA-Seq数据文件的子目录。我们称之为“WHOLE_SEQ-DATA_DIR”。我们公司这步的文件夹命名为“seq_data”</p>
<p>（2）使用sample_ID 命名 WHOLE_SEQ-DATA_DIR下的子目录的名称。我们称之为“SAMPLE_DIR”，我们公司这步的文件夹命名与样品名一致。</p>
<p>（3）使用sequence_library_ID 命名SAMPLE_DIR的子目录的名称。我们称之为“LIB_DIR”，与公司一致。</p>
<p>（4）RNA测序数据（FASTQ/ FASTA）文件存储在LIB_DIR中以其Run_ID作为文件名前缀，与公司一致。\<br>     SOAPfuse处理paired-end reads,所以前缀也应该含有双端reads序列号,如“Run_ID_1”、“Run_ID_2”。</p>
<p>（5）对 RNA-Seq 数据文件后缀没有要求，一般我们使用’fq.gz’ (fastq) 或者 ‘fa.gz’ (fasta)。</p>
<h5 id="Run-SOAPfuse"><a href="#Run-SOAPfuse" class="headerlink" title="Run SOAPfuse"></a>Run SOAPfuse</h5><p>SOAPfuse运行,需要准备配置文件,SOAPfuse将根据配置文件来运行。</p>
<p>1.检查配置文件:<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">$ <span class="built_in">cd</span> /PATH_WHERE_YOU_PUT_THE_PACKAGE/SOAPfuse-vX.X/config/，我们公司路径为/PUBLIC/software/HUMAN/SOAPfuse-v1.26/config</span><br><span class="line">$ less -S config.txt</span><br></pre></td></tr></table></figure></p>
<p>注意:</p>
<p>（1）所有以“＃”开头的行都被视为注释。</p>
<p>（2）值和参数名称由“=”隔开，’=’后为值，可以被修改。</p>
<p>（3）根据需要某些值可以设置为“是”或“否”，有些可以保留为默认值。</p>
<p>（4）检查每个参数的前缀</p>
<p>有五种前缀，分别是“DB”，“PG”，“PS”，“PD”和“PA”。</p>
<p>a.“DB”指数据库的信息。</p>
<p>b.“PG”是指程序的信息。</p>
<p>c.“PS”是指流程步骤的信息。</p>
<p>d.“PD”是指流程目录的信息。</p>
<p>e.“PA”是指参数的信息。</p>
<p>＃“DB”，“PG”，“PS”和“PD”与数据库相关，所以一旦这些参数准确地设置SOAPfuse能成功地运行。 “PA”与每个步骤的参数相关，并已设置为默认值，这样你就可以在你的第一次尝试时忽略它们，选择默认值。但是，“PA_all_fq_postfix”，它定义了RNA测序数据文件后缀，在运行前应该根据自己的RNA-Seq 文件设置参数值。</p>
<p>2.Run SOAPfuse</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">perl SOAPfuse-RUN.pl -c &lt;config_file&gt; -fd &lt;WHOLE_SEQ-DATA_DIR&gt; <span class="_">-l</span> &lt;sample_list&gt; -o &lt;out_directory&gt; [Options]</span><br><span class="line"></span><br><span class="line"><span class="comment">#####参数说明</span></span><br><span class="line">-c   运行流程运用的配制文件. &lt;required&gt;</span><br><span class="line">-fd  存储双端测序Reads文件的目录 . &lt;required&gt;       Sequenced Reads 可以为fastq 或者 fasta格式文件</span><br><span class="line"><span class="_">-l</span>   要处理的样本列表信息. &lt;required&gt;</span><br><span class="line">-o   结果存储目录</span><br><span class="line">-fs  想要开始的步骤. [1]</span><br><span class="line">-es  想要结束的步骤. [9]	第9步是 SOAPfuse pipeline的最后一步</span><br><span class="line">-h  显示帮助信息.</span><br></pre></td></tr></table></figure>
<h3 id="添加lncRNA癌基因数据库注释"><a href="#添加lncRNA癌基因数据库注释" class="headerlink" title="添加lncRNA癌基因数据库注释"></a>添加lncRNA癌基因数据库注释</h3><p>再lncRNA的差异分析中添加癌基因注释，本次使用lncRNA癌基因数据库为<a href="http://www.cuilab.cn/lncrnadisease" target="_blank" rel="external">http://www.cuilab.cn/lncrnadisease</a></p>
<p>注释脚本如下：<br><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#!/usr/bin/env python</span></span><br><span class="line"><span class="comment">#coding=utf-8</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">import</span> argparse</span><br><span class="line">parser=argparse.ArgumentParser(prog=<span class="string">'annotation'</span>,usage=<span class="string">'%(prog)s [opthions] [value]'</span>,description=<span class="string">'This program is used to annotate the differential expression genes!'</span>)</span><br><span class="line"></span><br><span class="line">parser.add_argument(<span class="string">'-D'</span>,<span class="string">'--diffgene'</span>,help=<span class="string">'the diffgene file'</span>)</span><br><span class="line">parser.add_argument(<span class="string">'-B'</span>,<span class="string">'--database'</span>,help=<span class="string">'the annotation file'</span>)</span><br><span class="line">parser.add_argument(<span class="string">'-O'</span>,<span class="string">'--outfile'</span>,help=<span class="string">'the output file'</span>)</span><br><span class="line">argv=vars(parser.parse_args())</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> argv[<span class="string">'diffgene'</span>] == <span class="keyword">None</span>:</span><br><span class="line">    <span class="keyword">raise</span> Exception(<span class="string">'You should provide the diffgene file!'</span>)</span><br><span class="line"><span class="keyword">if</span> argv[<span class="string">'diffgene'</span>] != <span class="keyword">None</span>:</span><br><span class="line">    diffgene=argv[<span class="string">'diffgene'</span>].strip()</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> argv[<span class="string">'database'</span>] == <span class="keyword">None</span>:</span><br><span class="line">    <span class="keyword">raise</span> Exception(<span class="string">'You should provide the annotation file!'</span>)</span><br><span class="line"><span class="keyword">if</span> argv[<span class="string">'database'</span>] != <span class="keyword">None</span>:</span><br><span class="line">    database=argv[<span class="string">'database'</span>].strip()</span><br><span class="line"></span><br><span class="line"><span class="keyword">if</span> argv[<span class="string">'outfile'</span>] == <span class="keyword">None</span>:</span><br><span class="line">    <span class="keyword">raise</span> Exception(<span class="string">'You should provide the output file!'</span>)</span><br><span class="line"><span class="keyword">if</span> argv[<span class="string">'outfile'</span>] != <span class="keyword">None</span>:</span><br><span class="line">    outfile=argv[<span class="string">'outfile'</span>].strip()</span><br><span class="line"></span><br><span class="line">genelist=&#123;&#125;</span><br><span class="line">Diffgene=open(diffgene)</span><br><span class="line">header=Diffgene.readline()</span><br><span class="line"><span class="keyword">for</span> eachline <span class="keyword">in</span> Diffgene:</span><br><span class="line">    eachlines=eachline.strip().split(<span class="string">'\t'</span>)</span><br><span class="line">    genelist[eachlines[<span class="number">1</span>]] = eachlines[<span class="number">0</span>]</span><br><span class="line">Diffgene.close()</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">Annotation=open(database)</span><br><span class="line">header=Annotation.readline()</span><br><span class="line">Outfile=open(outfile,<span class="string">'w'</span>)</span><br><span class="line">Outfile.write(header)</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> eachline <span class="keyword">in</span> Annotation:</span><br><span class="line">    eachlines=eachline.strip().split(<span class="string">'\t'</span>)</span><br><span class="line">    <span class="keyword">if</span> eachlines[<span class="number">0</span>] <span class="keyword">in</span> genelist.keys():</span><br><span class="line">        Outfile.write(genelist[eachlines[<span class="number">0</span>]] + <span class="string">"\t"</span> + eachline)</span><br><span class="line"></span><br><span class="line">Annotation.close()</span><br><span class="line">Outfile.close()</span><br></pre></td></tr></table></figure></p>

  </section>

  

<section class="post-comments">
  <div id="disqus_thread"></div>
  <script type="text/javascript">
      var disqus_shortname = 'youngguang'; 
      /* * * DON'T EDIT BELOW THIS LINE * * */
      (function() {
          var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
          dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
          (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
      })();
  </script>
  <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</section>


  
</article>


            <footer class="footer">

    <span class="footer__copyright">&copy; 2014-2015. | 由<a href="https://hexo.io/">Hexo</a>强力驱动 | 主题<a href="https://github.com/someus/huno">Huno</a></span>
    
</footer>
        </div>
    </div>

    <!-- js files -->
    <script src="/js/jquery.min.js"></script>
    <script src="/js/main.js"></script>
    <script src="/js/scale.fix.js"></script>
    

    

    <script type="text/javascript" src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript"> 
        $(document).ready(function(){
            MathJax.Hub.Config({ 
                tex2jax: {inlineMath: [['[latex]','[/latex]'], ['\\(','\\)']]} 
            });
        });
    </script>


    

    <script src="/js/awesome-toc.min.js"></script>
    <script>
        $(document).ready(function(){
            $.awesome_toc({
                overlay: true,
                contentId: "post-content",
            });
        });
    </script>


    <script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?5ed7e5c6530f805ef5b2f6d9c187ded6";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>

    
    <!--kill ie6 -->
<!--[if IE 6]>
  <script src="//letskillie6.googlecode.com/svn/trunk/2/zh_CN.js"></script>
<![endif]-->

</body>
</html>
