<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">

    

    <title>
      String和Cufflink拼接转录本的比较 | Sunshine&#39;s Life 
    </title>

    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
    
      <meta name="author" content="杨光">
    
    

    <meta name="description" content="StringTie应用网络流算法以及可选的从头组装来拼接转录本。在仿真数据集和真实数据集的分析结果中，相比于Cufflinks等方法，StringTie具有以下优势（1）拼接出更完整的转录本，（2）拼接处更准确的转录本，（3）更好的估计转录本的表达水平，（4）拼接速度更快">
<meta property="og:type" content="article">
<meta property="og:title" content="String和Cufflink拼接转录本的比较 | Sunshine's Life">
<meta property="og:url" content="http://yoursite.com/2017/04/13/String和Cufflink拼接转录本的比较/index.html">
<meta property="og:site_name" content="Sunshine's Life">
<meta property="og:description" content="StringTie应用网络流算法以及可选的从头组装来拼接转录本。在仿真数据集和真实数据集的分析结果中，相比于Cufflinks等方法，StringTie具有以下优势（1）拼接出更完整的转录本，（2）拼接处更准确的转录本，（3）更好的估计转录本的表达水平，（4）拼接速度更快">
<meta property="og:updated_time" content="2017-08-13T07:04:51.653Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="String和Cufflink拼接转录本的比较 | Sunshine's Life">
<meta name="twitter:description" content="StringTie应用网络流算法以及可选的从头组装来拼接转录本。在仿真数据集和真实数据集的分析结果中，相比于Cufflinks等方法，StringTie具有以下优势（1）拼接出更完整的转录本，（2）拼接处更准确的转录本，（3）更好的估计转录本的表达水平，（4）拼接速度更快">
    
    
    
      <link rel="icon" type="image/x-icon" href="/favicon.png">
    
    <link rel="stylesheet" href="/css/uno.css">
    <link rel="stylesheet" href="/css/highlight.css">
    <link rel="stylesheet" href="/css/archive.css">
    <link rel="stylesheet" href="/css/china-social-icon.css">

</head>
<body>

    <span class="mobile btn-mobile-menu">
        <i class="icon icon-list btn-mobile-menu__icon"></i>
        <i class="icon icon-x-circle btn-mobile-close__icon hidden"></i>
    </span>

    

<header class="panel-cover panel-cover--collapsed">


  <div class="panel-main">

  
    <div class="panel-main__inner panel-inverted">
    <div class="panel-main__content">

        

        <h1 class="panel-cover__title panel-title"><a href="/" title="link to homepage">Sunshine&#39;s Life</a></h1>
        <hr class="panel-cover__divider" />

        

        <div class="navigation-wrapper">

          <nav class="cover-navigation cover-navigation--primary">
            <ul class="navigation">

              
                
                <li class="navigation__item"><a href="/#blog" title="" class="blog-button">首页</a></li>
              
                
                <li class="navigation__item"><a href="/about" title="" class="">关于</a></li>
              
                
                <li class="navigation__item"><a href="/archive" title="" class="">归档</a></li>
              

            </ul>
          </nav>

          <!-- ----------------------------
To add a new social icon simply duplicate one of the list items from below
and change the class in the <i> tag to match the desired social network
and then add your link to the <a>. Here is a full list of social network
classes that you can use:

    icon-social-500px
    icon-social-behance
    icon-social-delicious
    icon-social-designer-news
    icon-social-deviant-art
    icon-social-digg
    icon-social-dribbble
    icon-social-facebook
    icon-social-flickr
    icon-social-forrst
    icon-social-foursquare
    icon-social-github
    icon-social-google-plus
    icon-social-hi5
    icon-social-instagram
    icon-social-lastfm
    icon-social-linkedin
    icon-social-medium
    icon-social-myspace
    icon-social-path
    icon-social-pinterest
    icon-social-rdio
    icon-social-reddit
    icon-social-skype
    icon-social-spotify
    icon-social-stack-overflow
    icon-social-steam
    icon-social-stumbleupon
    icon-social-treehouse
    icon-social-tumblr
    icon-social-twitter
    icon-social-vimeo
    icon-social-xbox
    icon-social-yelp
    icon-social-youtube
    icon-social-zerply
    icon-mail

-------------------------------->

<!-- add social info here -->



<nav class="cover-navigation navigation--social">
  <ul class="navigation">

  
      <!-- Github -->
      <li class="navigation__item">
        <a href="https://github.com/yangguang8112" title="Handsome boy on GitHub">
          <i class='icon icon-social-github'></i>
          <span class="label">GitHub</span>
        </a>
      </li>


    <!-- China social icon -->
    
    <!--
      <li class="navigation__item">
        <a href="" title="">
          <i class='icon cs-icon-douban'></i>
          <span class="label">Douban</span>
        </a>
      </li>
    -->
      <li class="navigation__item">
        <a href="http://weibo.com/1994YG?refer_flag=1001030101_&is_hot=1" title="一起下厕所">
          <i class='icon cs-icon-weibo'></i>
          <span class="label">Weibo</span>
        </a>
      </li>

  



  </ul>
</nav>

        </div>

      </div>

    </div>

    <div class="panel-cover--overlay"></div>
  </div>
</header>

    <div class="content-wrapper">
        <div class="content-wrapper__inner entry">
            

<article class="post-container post-container--single">

  <header class="post-header">
    
    <h1 class="post-title">String和Cufflink拼接转录本的比较</h1>

    

    <div class="post-meta">
      <time datetime="2017-04-13" class="post-meta__date date">2017-04-13</time> 

      <span class="post-meta__tags tags">

          

          
             &#8226; 标签:
            <font class="tags">
              <a class="tags-link" href="/tags/软件测试/">软件测试</a>
            </font>
          

      </span>
    </div>
    
    

  </header>

  <section id="post-content" class="article-content post">
    <p>StringTie应用网络流算法以及可选的从头组装来拼接转录本。在仿真数据集和真实数据集的分析结果中，相比于Cufflinks等方法，StringTie具有以下优势<br>（1）拼接出更完整的转录本，<br>（2）拼接处更准确的转录本，<br>（3）更好的估计转录本的表达水平，<br>（4）拼接速度更快<br><a id="more"></a></p>
<h3 id="StringTie流程"><a href="#StringTie流程" class="headerlink" title="StringTie流程"></a>StringTie流程</h3><p>stringtie.png</p>
<p><strong>Step1</strong>: （可选的）将reads组装成”Super-reads”。如果两个reads间有k-mer的overlap就进行延伸，指导两个方向都不再能延伸位置，最后组装成更长的序列，被称之为”Super-reads”。</p>
<p><strong>Step2</strong>：将readsmapping到参考基因组。<strong>StringTie+SR</strong>方法使用的是混合reads，即super-reads和unassembled-reads，而<strong>StringTie</strong>使用的是测序得到的所有reads。</p>
<p><strong>Step3</strong>: 经reads mapping，将reads聚集成cluster，对产生的每个cluster构建相应的可变剪切图。每个可变剪切代表基因所有可能的转录本异构体。</p>
<p><strong>Step4</strong>：从可变剪切图中识别有最高reads覆盖的路径，对该路径构建流网络。</p>
<p><strong>Step5</strong>：利用网络流算法来分配reads给转录本，使转录本覆盖的reads数目最多。</p>
<p><strong>Step6</strong>：去除剪切图中在Step5步骤使用过的reads，迭代进行Step4-Step5，直到没有路径可循。</p>
<h3 id="使用数据集"><a href="#使用数据集" class="headerlink" title="使用数据集"></a>使用数据集</h3><p><strong>（1）仿真数据集两套</strong>：<br>数据来源：UCSC Genome Browser<br>仿真方法：Flux Simulator<br>reads长度：150 million 75-bp paired-end reads</p>
<p>Sim-I: the fragment sizes follow an empirical distribution based on Illumina sequences,<br>Sim-II :the fragment sizes follow a parameterized normal distribution</p>
<p><strong>（2）真实数据集4套</strong>，其中3套链特异，1套链非特异的<br>数据来源：ENCODE</p>
<p>a. 链特异数据集：<br>Blood ：90 million 76-bp paired-end reads<br>Lung : 145 million 101-bp pairedend reads<br>Monocytes(单核细胞): 120 million 76-bp paired-end reads</p>
<p>b. 链非特异数据集：<br>nuclear RNA from a human kidney cell line(自己测)：~180 million 100-bp paired-end reads with an average fragment length of 177 bp\</p>
<h3 id="结果比较"><a href="#结果比较" class="headerlink" title="结果比较"></a>结果比较</h3><p><strong>1.性能比较：拼接转录本</strong></p>
<p>性能比较1.png</p>
<p>当考虑有reads部分覆盖（丰度低）和完全覆盖（丰度高）的转录本时，</p>
<p>与Cufflinks等方法相比，StringTie和StringTie+SR具有更高的灵敏度和更低的假阳性（1-精确度）</p>
<p>与StringTie方法相比，StringTie+SR的灵敏度和精确度均有稍微的提升，主要是因为StringTie+SR方法预先将reads拼接成更长的super-reads</p>
<p>性能比较2.png</p>
<p>当考虑有reads完全覆盖（高丰度）的转录本时，</p>
<p>与Cufflinks等方法相比，StringTie和StringTie+SR具有更高的灵敏度和更低的假阳性</p>
<p>与StringTie方法相比，StringTie+SR的灵敏度和精确度均有稍微的提升</p>
<p><strong>2.性能比较：计算转录本表达</strong></p>
<p>性能比较3.png</p>
<p>利用斯皮尔曼等级相关来计算识别转录本与真实转录本表达的相关性来评价转录本定量的准确性</p>
<p>在Sim-I和Sim-II数据集中，StringTie和StringTie+SR计算出的相关性均要高于Cufflinks等方法，说明这两种方法对转录本的定量要比Cufflinks方法更为准确</p>
<p><strong>3.识别转录本数量比较</strong></p>
<p>性能比较4.png</p>
<p>在四套真实数据集中，StringTie和StringTie+SR这两种方法要比Cufflinks等方法识别出更多数量的转录本</p>
<p><strong>4.计算的消耗</strong></p>
<table>
<thead>
<tr>
<th></th>
<th>StringTie+SR</th>
<th>StringTie</th>
<th>Cufflinks</th>
<th>Traph</th>
<th>Scripture</th>
<th>IsoLasso</th>
</tr>
</thead>
<tbody>
<tr>
<td>Sim-I</td>
<td>13</td>
<td>17</td>
<td>91</td>
<td>2885</td>
<td>1526</td>
<td>164</td>
</tr>
<tr>
<td>Sim-II</td>
<td>11</td>
<td>23</td>
<td>81</td>
<td>1515</td>
<td>1551</td>
<td>107</td>
</tr>
<tr>
<td>Kidney</td>
<td>39</td>
<td>76</td>
<td>941</td>
<td>n/a</td>
<td>4765</td>
<td>286</td>
</tr>
<tr>
<td>Blood</td>
<td>54</td>
<td>59</td>
<td>507</td>
<td>n/a</td>
<td>2377</td>
<td>305</td>
</tr>
<tr>
<td>Lung</td>
<td>59</td>
<td>53</td>
<td>1125</td>
<td>n/a</td>
<td>3159</td>
<td>326</td>
</tr>
<tr>
<td>Monocytes</td>
<td>33</td>
<td>35</td>
<td>411</td>
<td>n/a</td>
<td>2874</td>
<td>384</td>
</tr>
</tbody>
</table>
<p>（1）计算时间消耗：StringTie和StringTie+SR在6套数据集中转录本拼接时间均不足1h，而Cufflinks方法至少1.5h，最多将近20h</p>
<table>
<thead>
<tr>
<th></th>
<th>StringTie+SR</th>
<th>StringTie</th>
<th>Cufflinks</th>
<th>IsoLasso</th>
</tr>
</thead>
<tbody>
<tr>
<td>Sim-I</td>
<td>18.8</td>
<td>23.2</td>
<td>400.8</td>
<td>165.5</td>
</tr>
<tr>
<td>Sim-II</td>
<td>15.3</td>
<td>36.5</td>
<td>368.8</td>
<td>108.1</td>
</tr>
<tr>
<td>Kidney</td>
<td>48.2</td>
<td>95.1</td>
<td>2369.6</td>
<td>298.5</td>
</tr>
<tr>
<td>Blood</td>
<td>125</td>
<td>130.1</td>
<td>2166.5</td>
<td>305.2</td>
</tr>
<tr>
<td>Lung</td>
<td>142.5</td>
<td>135.8</td>
<td>4869</td>
<td>326.1</td>
</tr>
<tr>
<td>Monocytes</td>
<td>92.3</td>
<td>93.6</td>
<td>2491.7</td>
<td>383.9</td>
</tr>
</tbody>
</table>
<p>（2）CPU占用时间消耗：在6套数据集中，相比于Cufflinks等方法，StringTie和StringTie+SR方法均占用更少的CPU<br>|             |  StringTie+SR   | StringTie   | Cufflinks   | Scripture   | IsoLasso   |<br>|–|–|–|–|–|–|<br>| Kidney      | 11.7Gb          | 12Gb        | 26.6Gb      | 24.4Gb      | 20.4Gb     |<br>| Blood       | 4.7Gb           | 4.3Gb       | 6.4Gb       | 16Gb        | 13Gb       |<br>| Lung        | 6.9Gb           | 6.4Gb       | 7Gb         | 22.2Gb      | 8.8Gb      |<br>| Monocytes   | 1.6Gb           | 1.8Gb       | 6.6Gb       | 17.7Gb      | 13.2Gb     |</p>
<p>（3）最大内存消耗：真实数据集中，StringTie和StringTie+SR最多消耗12G左右，而Cufflinks消耗多达26G</p>
<p>综合以上消耗，拼接转录本时，在计算时间、CPU占有时间以及最大内存消耗方面，StringTie和StringTie+SR相比于Cufflinks具有更少的消耗</p>
<h3 id="测试结果比较"><a href="#测试结果比较" class="headerlink" title="测试结果比较"></a>测试结果比较</h3><p>使用的是〖P2016080179〗浙医二院6个人lncRNA测序及分析软件RNAseq-Homosapiens-TY的开发项目经Tophat比对后产生的bam文件</p>
<p><strong>1. 计算消耗的比较</strong></p>
<p>| Sample  | Bam    | cpu   || Mem(GBs)  || io  || vmem      || maxvmem  ||<br>|         |        | STie     | Cuff      | STie      | Cuff      | STie   | Cuff   | STie       | Cuff     | STie       | Cuff     |<br>|–|–|–|–|–|–|–|–|–|–|–|–|<br>| H1      | 4.7G   | 0:09:57  | 10:01:16  | 278.41    | 56471.39  | 5.11   | 21.53  | 570.449M   | 1.506G   | 947.520M   | 3.396G   |<br>| H2      | 5.2G   | 0:14:58  | 11:31:45  | 410.93    | 77337.35  | 5.50   | 30.23  | 588.285M   | 1.841G   | 716.078M   | 3.240G   |<br>| H3      | 5.5G   | 0:15:09  | 12:02:42  | 427.41    | 78301.86  | 5.92   | 29.15  | 575.176M   | 1.803G   | 740.000M   | 3.744G   |<br>| N1      | 4.7G   | 0:10:05  | 9:45:38   | 278.36    | 52799.4   | 5.05   | 23.25  | 613.043M   | 1.929G   | 896.539M   | 2.937G   |<br>| N2      | 5.9G   | 0:09:47  | 11:21:45  | 269.17    | 109620.8  | 4.99   | 27.51  | 572.027M   | 2.682G   | 748.594M   | 4.227G   |<br>| N3      | 5.8G   | 0:10:23  | 11:10:59  | 308.73    | 85997.38  | 6.18   | 24.13  | 563.809M   | 2.196G   | 1.157G     | 3.636G   |<br>| H1      | 4.7G   | 0:09:57  | 10:01:16  | 278.41    | 56471.39  | 5.11   | 21.53  | 570.449M   | 1.506G   | 947.520M   | 3.396G   |</p>
<p>平均消耗：</p>
<table>
<thead>
<tr>
<th></th>
<th>CPU</th>
<th>Mem</th>
<th>io</th>
<th>Vmem</th>
<th>Maxvmem</th>
</tr>
</thead>
<tbody>
<tr>
<td>StringTie</td>
<td>11.49min</td>
<td>328.83G</td>
<td>5.5G</td>
<td>580M</td>
<td>846M</td>
</tr>
<tr>
<td>Cufflinks</td>
<td>10.85h</td>
<td>76754G</td>
<td>25.83G</td>
<td>1.95G</td>
<td>3.5G</td>
</tr>
</tbody>
</table>
<p>由此可发现，StringTie拼接转录本消耗要比Cufflinks消耗的更少</p>
<p><strong>2. 拼接转录本数量的比较</strong></p>
<table>
<thead>
<tr>
<th>Sample</th>
<th>Size</th>
<th>StringTie</th>
<th>Cufflinks</th>
</tr>
</thead>
<tbody>
<tr>
<td>H1</td>
<td>4.7G</td>
<td>86735</td>
<td>208518</td>
</tr>
<tr>
<td>H2</td>
<td>5.2G</td>
<td>155010</td>
<td>249061</td>
</tr>
<tr>
<td>H3</td>
<td>5.5G</td>
<td>132406</td>
<td>227071</td>
</tr>
<tr>
<td>N1</td>
<td>4.7G</td>
<td>95246</td>
<td>209199</td>
</tr>
<tr>
<td>N2</td>
<td>5.9G</td>
<td>117743</td>
<td>215553</td>
</tr>
<tr>
<td>N3</td>
<td>5.8G</td>
<td>93459</td>
<td>208438</td>
</tr>
<tr>
<td>Merge</td>
<td></td>
<td>204852</td>
<td>306562</td>
</tr>
</tbody>
</table>
<p>测试StringTie拼接出来的转录本数量远少于Cufflinks拼接的数量，后续还要对两种方法拼接转录本的准确性进行验证。</p>
<p><strong>3. 拼接转录本长度比较</strong></p>
<p>长度.jpg</p>
<h3 id="使用的脚本"><a href="#使用的脚本" class="headerlink" title="使用的脚本"></a>使用的脚本</h3><ol>
<li>StringTie</li>
</ol>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">stringtie accepted_hits.bam -G Homo_sapiens.GRCh38.84.lnc.exon.gtf -B -o H1_transcripts.gtf</span><br><span class="line"></span><br><span class="line">stringtie --merge merge.list -p 8 -G Homo_sapiens.GRCh38.84.lnc.exon.gtf -o stringtie_merge.gtf -B -t</span><br></pre></td></tr></table></figure>
<ol>
<li>Cufflinks</li>
</ol>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"></span><br><span class="line">/PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/softwares/cufflinks-2.2.1_patched/cufflinks -g Homo_sapiens.GRCh38.84.lnc.exon.gtf -u --library-type fr-firststrand -o N1 accepted_hits.bam</span><br><span class="line"></span><br><span class="line">/PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/softwares/cufflinks-2.2.1_patched//cuffmerge -o merge -p 8 -g Homo_sapiens.GRCh38.84.lnc.exon.gtf <span class="_">-s</span> Homo_sapiens.GRCh38.dna.primary_assembly.fa merge.list</span><br></pre></td></tr></table></figure>
<ol>
<li>统计结果</li>
</ol>
<pre><code class="r">
 StatInfor &lt;- <span class="keyword">function</span>(path)
 {
    setwd(path)                                          
    options(StringsAsFactors=<span class="literal">FALSE</span>)
    f = list.files()
    s = setdiff(grep(<span class="string">'_assembly.sh.'</span>,f),grep(<span class="string">'.png'</span>,f))
    a = sapply(f[s],<span class="keyword">function</span>(x){read.delim2(x,header=<span class="literal">FALSE</span>,sep=<span class="string">"\t"</span>)[[<span class="number">2</span>]]})
    temp &lt;- c()
    <span class="keyword">for</span>(i <span class="keyword">in</span> <span class="number">1</span>:length(s)){
        b = a[[i]][length(a[[i]])]
        u = unlist(strsplit(unlist(strsplit(as.character(b),<span class="string">","</span>)),<span class="string">" "</span>))
        z = c(substr(f[s][i],<span class="number">1</span>,<span class="number">2</span>),
            gsub(<span class="string">"cpu="</span>,<span class="string">""</span>,u[<span class="number">22</span>]),
            gsub(<span class="string">"mem="</span>,<span class="string">""</span>,u[<span class="number">24</span>]),
            gsub(<span class="string">"io="</span>,<span class="string">""</span>,u[<span class="number">27</span>]),
            gsub(<span class="string">"vmem="</span>,<span class="string">""</span>,u[<span class="number">29</span>]),
            gsub(<span class="string">"maxvmem="</span>,<span class="string">""</span>,u[<span class="number">31</span>])
            )
    temp &lt;- rbind(temp,z)
     }
    <span class="keyword">return</span>(temp)
}

args&lt;-commandArgs(<span class="literal">T</span>)
<span class="keyword">source</span>(<span class="string">"statFun.R"</span>)
STie = StatInfor(args[<span class="number">1</span>])
Cuff = StatInfor(args[<span class="number">2</span>])
Stat = data.frame(STie[,<span class="number">1</span>:<span class="number">2</span>],Cuff[,<span class="number">2</span>],STie[,<span class="number">3</span>],Cuff[,<span class="number">3</span>],STie[,<span class="number">4</span>],Cuff[,<span class="number">4</span>],STie[,<span class="number">5</span>],Cuff[,<span class="number">5</span>],STie[,<span class="number">6</span>],Cuff[,<span class="number">6</span>])
setwd(<span class="string">'/PROJ/test/StringTie_Cufflinks_Test'</span>)
colnames(Stat) &lt;- c(<span class="string">"Sample"</span>,paste(c(<span class="string">"STie"</span>,<span class="string">"Cuff"</span>),rep(c(<span class="string">"cpu"</span>,<span class="string">"mem"</span>,<span class="string">"io"</span>,<span class="string">"vmem"</span>,<span class="string">"maxvmem"</span>),each=<span class="number">2</span>),sep=<span class="string">"_"</span>))
write.table(Stat,<span class="string">"Stat.txt"</span>,col.names=<span class="literal">T</span>,quote=<span class="literal">F</span>,sep=<span class="string">"\t"</span>,row.names=<span class="literal">FALSE</span>)
</code></pre>

  </section>

  

<section class="post-comments">
  <div id="disqus_thread"></div>
  <script type="text/javascript">
      var disqus_shortname = 'youngguang'; 
      /* * * DON'T EDIT BELOW THIS LINE * * */
      (function() {
          var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
          dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
          (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
      })();
  </script>
  <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</section>


  
</article>


            <footer class="footer">

    <span class="footer__copyright">&copy; 2014-2015. | 由<a href="https://hexo.io/">Hexo</a>强力驱动 | 主题<a href="https://github.com/someus/huno">Huno</a></span>
    
</footer>
        </div>
    </div>

    <!-- js files -->
    <script src="/js/jquery.min.js"></script>
    <script src="/js/main.js"></script>
    <script src="/js/scale.fix.js"></script>
    

    

    <script type="text/javascript" src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript"> 
        $(document).ready(function(){
            MathJax.Hub.Config({ 
                tex2jax: {inlineMath: [['[latex]','[/latex]'], ['\\(','\\)']]} 
            });
        });
    </script>


    

    <script src="/js/awesome-toc.min.js"></script>
    <script>
        $(document).ready(function(){
            $.awesome_toc({
                overlay: true,
                contentId: "post-content",
            });
        });
    </script>


    <script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?5ed7e5c6530f805ef5b2f6d9c187ded6";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>

    
    <!--kill ie6 -->
<!--[if IE 6]>
  <script src="//letskillie6.googlecode.com/svn/trunk/2/zh_CN.js"></script>
<![endif]-->

</body>
</html>
