<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">

    

    <title>
      lncRNA筛选 | Sunshine&#39;s Life 
    </title>

    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
    
      <meta name="author" content="杨光">
    
    

    <meta name="description" content="lncRNA的筛选是lncRNA分析流程中的关键步骤梳理一下流程中的筛选步骤,还有4种编码潜能预测软件">
<meta property="og:type" content="article">
<meta property="og:title" content="lncRNA筛选 | Sunshine's Life">
<meta property="og:url" content="http://yoursite.com/2016/11/28/lncRNA筛选/index.html">
<meta property="og:site_name" content="Sunshine's Life">
<meta property="og:description" content="lncRNA的筛选是lncRNA分析流程中的关键步骤梳理一下流程中的筛选步骤,还有4种编码潜能预测软件">
<meta property="og:updated_time" content="2017-08-12T12:11:01.664Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="lncRNA筛选 | Sunshine's Life">
<meta name="twitter:description" content="lncRNA的筛选是lncRNA分析流程中的关键步骤梳理一下流程中的筛选步骤,还有4种编码潜能预测软件">
    
    
    
      <link rel="icon" type="image/x-icon" href="/favicon.png">
    
    <link rel="stylesheet" href="/css/uno.css">
    <link rel="stylesheet" href="/css/highlight.css">
    <link rel="stylesheet" href="/css/archive.css">
    <link rel="stylesheet" href="/css/china-social-icon.css">

</head>
<body>

    <span class="mobile btn-mobile-menu">
        <i class="icon icon-list btn-mobile-menu__icon"></i>
        <i class="icon icon-x-circle btn-mobile-close__icon hidden"></i>
    </span>

    

<header class="panel-cover panel-cover--collapsed">


  <div class="panel-main">

  
    <div class="panel-main__inner panel-inverted">
    <div class="panel-main__content">

        

        <h1 class="panel-cover__title panel-title"><a href="/" title="link to homepage">Sunshine&#39;s Life</a></h1>
        <hr class="panel-cover__divider" />

        

        <div class="navigation-wrapper">

          <nav class="cover-navigation cover-navigation--primary">
            <ul class="navigation">

              
                
                <li class="navigation__item"><a href="/#blog" title="" class="blog-button">首页</a></li>
              
                
                <li class="navigation__item"><a href="/about" title="" class="">关于</a></li>
              
                
                <li class="navigation__item"><a href="/archive" title="" class="">归档</a></li>
              

            </ul>
          </nav>

          <!-- ----------------------------
To add a new social icon simply duplicate one of the list items from below
and change the class in the <i> tag to match the desired social network
and then add your link to the <a>. Here is a full list of social network
classes that you can use:

    icon-social-500px
    icon-social-behance
    icon-social-delicious
    icon-social-designer-news
    icon-social-deviant-art
    icon-social-digg
    icon-social-dribbble
    icon-social-facebook
    icon-social-flickr
    icon-social-forrst
    icon-social-foursquare
    icon-social-github
    icon-social-google-plus
    icon-social-hi5
    icon-social-instagram
    icon-social-lastfm
    icon-social-linkedin
    icon-social-medium
    icon-social-myspace
    icon-social-path
    icon-social-pinterest
    icon-social-rdio
    icon-social-reddit
    icon-social-skype
    icon-social-spotify
    icon-social-stack-overflow
    icon-social-steam
    icon-social-stumbleupon
    icon-social-treehouse
    icon-social-tumblr
    icon-social-twitter
    icon-social-vimeo
    icon-social-xbox
    icon-social-yelp
    icon-social-youtube
    icon-social-zerply
    icon-mail

-------------------------------->

<!-- add social info here -->



<nav class="cover-navigation navigation--social">
  <ul class="navigation">

  
      <!-- Github -->
      <li class="navigation__item">
        <a href="https://github.com/yangguang8112" title="Handsome boy on GitHub">
          <i class='icon icon-social-github'></i>
          <span class="label">GitHub</span>
        </a>
      </li>


    <!-- China social icon -->
    
    <!--
      <li class="navigation__item">
        <a href="" title="">
          <i class='icon cs-icon-douban'></i>
          <span class="label">Douban</span>
        </a>
      </li>
    -->
      <li class="navigation__item">
        <a href="http://weibo.com/1994YG?refer_flag=1001030101_&is_hot=1" title="一起下厕所">
          <i class='icon cs-icon-weibo'></i>
          <span class="label">Weibo</span>
        </a>
      </li>

  



  </ul>
</nav>

        </div>

      </div>

    </div>

    <div class="panel-cover--overlay"></div>
  </div>
</header>

    <div class="content-wrapper">
        <div class="content-wrapper__inner entry">
            

<article class="post-container post-container--single">

  <header class="post-header">
    
    <h1 class="post-title">lncRNA筛选</h1>

    

    <div class="post-meta">
      <time datetime="2016-11-28" class="post-meta__date date">2016-11-28</time> 

      <span class="post-meta__tags tags">

          

          
             &#8226; 标签:
            <font class="tags">
              <a class="tags-link" href="/tags/生物信息学/">生物信息学</a>
            </font>
          

      </span>
    </div>
    
    

  </header>

  <section id="post-content" class="article-content post">
    <p>lncRNA的筛选是lncRNA分析流程中的关键步骤<br>梳理一下流程中的筛选步骤,还有4种编码潜能预测软件<br><a id="more"></a></p>
<h3 id="链方向，exon，长度筛选"><a href="#链方向，exon，长度筛选" class="headerlink" title="链方向，exon，长度筛选"></a>链方向，exon，长度筛选</h3><p>1.去除无链方向的转录本<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">awk <span class="string">'&#123;if ($7 != ".")&#123;print $0&#125;&#125;'</span> merged.gtf &gt; step0_stranded_merged.gtf</span><br></pre></td></tr></table></figure></p>
<p>2.去除但外显子及长度筛选<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">python /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/exon_length_filter.py --gtf step0_stranded_merged.gtf --out_dir outdir --single_exon no/yes</span><br><span class="line"><span class="comment">#结果文件为：exonfilter:step1_exon_filter.gtf ,长度筛选：step2_length_filter.gtf；其中single_exon参数，植物选yes，动物选no；</span></span><br><span class="line"><span class="comment">#同时输出assembled_tr_info.json拼接转录本信息的json文件，用于后续提取转录本信息</span></span><br></pre></td></tr></table></figure></p>
<h3 id="已知注释筛选"><a href="#已知注释筛选" class="headerlink" title="已知注释筛选"></a>已知注释筛选</h3><p>1.去除有已知数据库注释的lncRNA（如物种无已知数据库注释，则不做该部分筛除）:<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">/PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/softwares/cufflinks-2.2.1_patched//cuffcompare -T -C -r gencode_v24.gtf -R step2_length_filter.gtf -o lnc_database_compare</span><br><span class="line"><span class="comment">#-r 已知lnc数据库中的注释文件；-R 通过exon及长度筛选的转录本注释文件 -o 输出文件前缀 </span></span><br><span class="line">python /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/annotated_transcript_filter.py --input_gtf step2_length_filter.gtf --compared_gtf lnc_database_compare.combined.gtf --output_gtf tmp_step3_1_lncRNA_filter.gtf --ref_gtf gencode_v24.gtf --compare_flag lncRNA</span><br><span class="line"><span class="comment">#-input_gtf 通过exon及长度筛选的转录本注释文件 ;--compared_gtf cuffcompare输出的所有转录本的combine注释文件 ；--ref_gtf 已知lnc数据库中的注释文件 ； --compare_flag 本次筛除的注释转录本类型</span></span><br></pre></td></tr></table></figure></p>
<p>2.去除参考基因组中的lncRNA<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">/PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/softwares/cufflinks-2.2.1_patched//cuffcompare -T -C -r lncRNA_filter/ref/lncRNA.gtf -R tmp_step3_1_lncRNA_filter.gtf -o ref_lnc_compare</span><br><span class="line"><span class="comment">#-r 参考基因组中lncRNA注释文件；-R 已筛除数据库中lncRNA的转录本注释文件 -o 输出文件前缀</span></span><br><span class="line">python /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/annotated_transcript_filter.py --input_gtf tmp_step3_1_lncRNA_filter.gtf --compared_gtf ref_lnc_compare.combined.gtf --output_gtf step3_1_lncRNA_filter.gtf --ref_gtf lncRNA_filter/ref/lncRNA.gtf --compare_flag lncRNA</span><br><span class="line"><span class="comment">#-input_gtf 已筛除数据库中lncRNA的转录本注释文件 ;--compared_gtf cuffcompare输出的所有转录本的combine注释文件 ；--ref_gtf 参考基因组中lncRNA注释文件 ； --compare_flag 本次筛除的注释转录本类型</span></span><br></pre></td></tr></table></figure></p>
<p>3.去除参考基因组已注释的其他RNA<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">/PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/softwares/cufflinks-2.2.1_patched//cuffcompare -T -C -r lncRNA_filter/ref/other_ncRNA.gtf -R step3_1_lncRNA_filter.gtf -o other_ncRNA_compare</span><br><span class="line">python /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/annotated_transcript_filter.py --input_gtf step3_1_lncRNA_filter.gtf --compared_gtf other_ncRNA_compare.combined.gtf --output_gtf step3_2_other_ncRNA_filter.gtf --compare_flag other_ncRNA</span><br></pre></td></tr></table></figure></p>
<p>4.通过merge注释的classcode进行已知mrna的注释去除<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/annotated_transcript_filter.py --input_gtf step3_2_other_ncRNA_filter.gtf --compared_gtf step0_stranded_merged.gtf --output_gtf step3_3_mRNA_filter.gtf --compare_flag mRNA</span><br></pre></td></tr></table></figure></p>
<h3 id="对所有准备用于分析的转录本进行定量"><a href="#对所有准备用于分析的转录本进行定量" class="headerlink" title="对所有准备用于分析的转录本进行定量"></a>对所有准备用于分析的转录本进行定量</h3><p>1.准备<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">cat ref_mRNA_gtf step3_3_mRNA_filter_gtf annotated_lncRNA.gtf &gt; step4_exp_<span class="built_in">test</span>_gtf</span><br><span class="line"><span class="comment">#将现阶段所有可用的RNA注释文件cat到一起，用于后续定量; annotated_lncRNA.gtf整合了已知数据库及参考基因组中注释的已知lncRNA</span></span><br></pre></td></tr></table></figure></p>
<p>2.单样品定量<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">/PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/softwares/cufflinks-2.2.1_patched//cuffquant --library-type fr-firststrand -o outdir step4_exp_test.gtf sample1.bam -p 2</span><br><span class="line"><span class="comment">#输出文件为abundances.cxb的二进制文件</span></span><br></pre></td></tr></table></figure></p>
<p>3.cuffnorm进行cuffquant定量结果的整合<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">/PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/softwares/cufflinks-2.2.1_patched//cuffnorm --library-type fr-firststrand -o cuffnorm \</span><br><span class="line"> -L sample1,samplen \</span><br><span class="line"> step4_exp_test.gtf \</span><br><span class="line">sample1/abundances.cxb </span><br><span class="line">samplen/abundances.cxb   --num-threads 2</span><br><span class="line"><span class="comment">#-o 定量结果输出路径（结果文件isoforms,gene,cds等的fpkm_table及count_table）,lncRNA分析中应用的isoforms.fpkm_table </span></span><br><span class="line"><span class="comment">#-L  样品名列表（“,”分隔） </span></span><br><span class="line">python /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/expression_filter.py --exp_table isoforms.fpkm_table --gtf step4_exp_test.gtf --tr_info assembled_tr_info.json --output step4_exp_filter.gtf</span><br><span class="line"><span class="comment">#多外显子过滤掉所有样品fpkm&lt;0.5的转录本；单外显子过滤掉所有样品fpkm&lt;2的转录本；</span></span><br><span class="line"><span class="comment">#输出结果：1. step4_exp_filter.gtf；2.exp_annotated_lncRNA.gtf(过滤掉低表达的已知lnc)</span></span><br><span class="line">perl /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/gtf2trbed.pl step4_exp_filter.gtf step4_exp_filter.bed   <span class="comment">#gtf to bed</span></span><br><span class="line">perl /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/lncRNA_v1/public/extractTranscriptsfromFA.pl step4_exp_filter.gtf genome.fa step4_exp_filter.fa  <span class="comment">#提取转录本序列文件</span></span><br><span class="line">perl /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/seq_split_v1.pl step4_exp_filter.fa step4_exp_filter 200 step4_exp_filter_split_fa </span><br><span class="line"><span class="comment">#将过滤了低表达转录本的序列以200条作为分隔，输出到step4_exp_filter_split_fa目录下。结果为step4_exp_filter_1.fa，step4_exp_filter_2.fa....step4_exp_filter_n.fa</span></span><br></pre></td></tr></table></figure></p>
<h2 id="编码潜能预测软件"><a href="#编码潜能预测软件" class="headerlink" title="编码潜能预测软件"></a>编码潜能预测软件</h2><h3 id="PFAM"><a href="#PFAM" class="headerlink" title="PFAM"></a>PFAM</h3><h4 id="分析步骤"><a href="#分析步骤" class="headerlink" title="分析步骤"></a>分析步骤</h4><p>这一步是拆分的<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">perl /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/Filter/PFAM/seq2protein.pl exp_filter_1.fa protein.fa</span><br><span class="line"><span class="comment">#将转录本序列转换为蛋白质序列</span></span><br><span class="line">perl /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/Filter/PFAM/pfam_scan.pl \</span><br><span class="line">-fasta protein.fa \</span><br><span class="line">-dir /PUBLIC/database/Common/PFAM \</span><br><span class="line">-outfile pfam_scan.out \</span><br><span class="line">-pfamB -cpu 2</span><br><span class="line"><span class="comment">#将蛋白质序列与pfam数据库比对,输出结果为pfam_scan.out</span></span><br></pre></td></tr></table></figure></p>
<p>处理pfam的输出文件,得到无编码潜能的转录本id列表<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">cat pfam_scan.out |sed <span class="_">-e</span> <span class="string">'/^#/d'</span> <span class="_">-e</span> <span class="string">'/^$/d'</span> |awk <span class="string">'OFS="\t"&#123;print ($1,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15)&#125;'</span> &gt; pfam_scan.out.temp</span><br><span class="line">cat /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/lncRNA_v1/Filter/PFAM/bin/name.xls pfam_scan.out.temp &gt; PFAM.result.xls</span><br><span class="line">python /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/lncRNA_v1/Filter/PFAM/bin/select_pfam_id.py exp_filter.gtf /pfam_scan.out.temp PFAM.id.txt</span><br><span class="line">rm pfam_scan.out.temp</span><br></pre></td></tr></table></figure></p>
<h3 id="CPC"><a href="#CPC" class="headerlink" title="CPC"></a>CPC</h3><h4 id="分析步骤-1"><a href="#分析步骤-1" class="headerlink" title="分析步骤"></a>分析步骤</h4><p>这步是拆分的<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">perl /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/CPC/runCPC.pl -seq exp_filter_1.fa -db eu -strand plus -outdir outdir</span><br></pre></td></tr></table></figure></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#runCPC.pl参数help信息</span></span><br><span class="line">perl /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/CPC/runCPC.pl --help</span><br><span class="line"><span class="comment">## </span></span><br><span class="line">Run Coding Potential Calculator program.</span><br><span class="line"></span><br><span class="line">Options: </span><br><span class="line">	-seq	&lt;s&gt;:	Fasta formated sequences to be calculated.</span><br><span class="line">	-db	&lt;s&gt;:	Protein database to search for. &#123;<span class="string">"nr"</span>,<span class="string">"eu"</span> or <span class="string">"sp"</span>&#125;;</span><br><span class="line">			  nr: the whole NCBI protein database;</span><br><span class="line">			  eu: the NCBI eukaryotes` protein database;</span><br><span class="line">			  sp: the SwissProt database;</span><br><span class="line">			  Not needed <span class="keyword">if</span> blast result provided.</span><br><span class="line">	-blast	&lt;s&gt;:	Blast result agaist known protein database.</span><br><span class="line">	-bfmt	&lt;s&gt;:	Blast format. &#123;blast, blastxml or blasttable&#125;.</span><br><span class="line">			  Corespand to 0,5,6 blast+ outfmt option.</span><br><span class="line">	-strand &lt;s&gt;:	Which strand of sequences <span class="keyword">for</span> coding potential analysis.</span><br><span class="line">			  <span class="string">"both"</span>, <span class="string">"plus"</span> or <span class="string">"minus"</span>.</span><br><span class="line">	-outdir	&lt;s&gt;:	Output directory.</span><br><span class="line">	-help	&lt;s&gt;:	help.</span><br><span class="line">Usage:</span><br><span class="line">	1. perl /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/CPC/runCPC.pl -seq &lt;..&gt; -db &lt;..&gt; -strand &lt;..&gt; -outdir &lt;..&gt;</span><br><span class="line">	2. perl /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/CPC/runCPC.pl -seq &lt;..&gt; -blast &lt;..&gt; -bfmt &lt;..&gt; -outdir</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">##</span></span><br></pre></td></tr></table></figure>
<p>这一步的输出文件名前缀为fa的文件名,总共输出四个文件:<br>blast.txt<br>cpc.txt<br>evidence.homo<br>evidence.orf<br>其中cpc.txt为编码潜能预测结果</p>
<p>接下来同样是结果文件处理,获得非编码转录本id<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">cat  /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/lncRNA_v1/Filter/CPC/bin/name.xls cpc.txt &gt; CPC.result.xls</span><br><span class="line">awk <span class="string">'&#123;if($3 == "noncoding")&#123;print $1&#125;&#125;'</span> CPC.result.xls &gt; CPC.id.txt</span><br></pre></td></tr></table></figure></p>
<h3 id="CNCI"><a href="#CNCI" class="headerlink" title="CNCI"></a>CNCI</h3><h4 id="分析步骤-2"><a href="#分析步骤-2" class="headerlink" title="分析步骤"></a>分析步骤</h4><p>拆分<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/softwares/CNCI_package/CNCI.py <span class="_">-f</span> exp_filter_1.fa -o out_dir -m ve -p 1</span><br></pre></td></tr></table></figure></p>
<p>CNCI的help信息<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br></pre></td><td class="code"><pre><span class="line">/PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/softwares/CNCI_package/CNCI.py -help</span><br><span class="line">Usage: CNCI.py [options]</span><br><span class="line"></span><br><span class="line">Options:</span><br><span class="line">  -h, --help            show this <span class="built_in">help</span> message and <span class="built_in">exit</span></span><br><span class="line">  <span class="_">-f</span> input files, --file=input files</span><br><span class="line">                        enter your transcript (sequence or gtf)</span><br><span class="line">  -o output files, --out=output files</span><br><span class="line">                        assign your output file</span><br><span class="line">  -p prallel numbers, --parallel=prallel numbers</span><br><span class="line">                        please enter your specified speed ratio</span><br><span class="line">  -m model types, --model=model types</span><br><span class="line">                        please enter your specified classification model</span><br><span class="line">  -g, --gtf             please enter your gtf files</span><br><span class="line">  <span class="_">-d</span> DIRECTORY, --directory=DIRECTORY</span><br><span class="line">                        <span class="keyword">if</span> your input file is gtf <span class="built_in">type</span> please enter RefGenome</span><br><span class="line">                        directory</span><br></pre></td></tr></table></figure></p>
<p>输出文件为CNCI.index,包含是否编码的信息</p>
<p>merge获得结果<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/merge_cnci.py cnci_dir</span><br></pre></td></tr></table></figure></p>
<h3 id="phyloCSF"><a href="#phyloCSF" class="headerlink" title="phyloCSF"></a>phyloCSF</h3><h4 id="分析步骤-3"><a href="#分析步骤-3" class="headerlink" title="分析步骤"></a>分析步骤</h4><h5 id="prepare"><a href="#prepare" class="headerlink" title="prepare"></a>prepare</h5><p>以染色体拆分<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">python /PUBLIC/software/RNA/galaxy/galaxy-dist2/galaxy-dist/tools/maf/interval_maf_to_merged_fasta.py <span class="_">-d</span> hg38 -c 1 <span class="_">-s</span> 2 <span class="_">-e</span> 3 -S 6 -m chr1.maf -i split_bed//chr1.bed -t user -o split_maf/extract.chr1.maf</span><br><span class="line"><span class="comment">#chr1.maf是之前准备好的maf文件</span></span><br><span class="line">sed <span class="_">-e</span> <span class="string">'s/hg38./hg38|/g'</span> split_maf/extract.chr1.maf &gt;split_maf/rename.extract.chr1.maf</span><br><span class="line">awk <span class="string">'&#123;print $4"\t"$1"("$6"):"$2"-"$3&#125;'</span> split_bed//chr1.bed &gt; split_bed//chr1.table</span><br><span class="line">perl /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/Filter/phyloCSF/bin/pre_phylocsf.modi.pl split_bed//chr1.table hg38 split_maf/rename.extract.chr1.maf split_fa</span><br><span class="line">perl /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/generate_phyloCSF_v2.pl -split_fa_dir split_fa -table split_bed//chr1.table -outdir split_script/chr1/ -maf_num hg38</span><br><span class="line">python /PUBLIC/<span class="built_in">source</span>/RNA/lncRNA/filter/version5/phylocsf_split.py split_script/chr1//script/CSF.sh 200 split_script/chr1//script/split/</span><br></pre></td></tr></table></figure></p>
<p>/TJPROJ1/RNA/ncRNA201704/lncRNA/C101SC16120656_ren_lnc_2/lncRNA_filter/filter/step5_phyloCSF/split_script</p>

  </section>

  

<section class="post-comments">
  <div id="disqus_thread"></div>
  <script type="text/javascript">
      var disqus_shortname = 'youngguang'; 
      /* * * DON'T EDIT BELOW THIS LINE * * */
      (function() {
          var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
          dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
          (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
      })();
  </script>
  <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</section>


  
</article>


            <footer class="footer">

    <span class="footer__copyright">&copy; 2014-2015. | 由<a href="https://hexo.io/">Hexo</a>强力驱动 | 主题<a href="https://github.com/someus/huno">Huno</a></span>
    
</footer>
        </div>
    </div>

    <!-- js files -->
    <script src="/js/jquery.min.js"></script>
    <script src="/js/main.js"></script>
    <script src="/js/scale.fix.js"></script>
    

    

    <script type="text/javascript" src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript"> 
        $(document).ready(function(){
            MathJax.Hub.Config({ 
                tex2jax: {inlineMath: [['[latex]','[/latex]'], ['\\(','\\)']]} 
            });
        });
    </script>


    

    <script src="/js/awesome-toc.min.js"></script>
    <script>
        $(document).ready(function(){
            $.awesome_toc({
                overlay: true,
                contentId: "post-content",
            });
        });
    </script>


    <script>
var _hmt = _hmt || [];
(function() {
  var hm = document.createElement("script");
  hm.src = "//hm.baidu.com/hm.js?5ed7e5c6530f805ef5b2f6d9c187ded6";
  var s = document.getElementsByTagName("script")[0]; 
  s.parentNode.insertBefore(hm, s);
})();
</script>

    
    <!--kill ie6 -->
<!--[if IE 6]>
  <script src="//letskillie6.googlecode.com/svn/trunk/2/zh_CN.js"></script>
<![endif]-->

</body>
</html>
